### ISSUE TITLE
[BUG] EVENT ID 10129 After Chapter 1.

### ISSUE THREAD

**Issue by marceloalexei** [cisagov/LME#140]
## Describe the bug
Hello all. After completing chapter 1 and 2, I get an event ID 10129, stating the following :
"The WS-Management client is not listening for pushed events because there was a failure binding to the URL (...) in HTTP.SYS.
It then asks me to use netsh http and make sure that URL is set to Network Service.
Error code given is 5: %%5 apparently.

The reason I'm not giving the URL is because it doesn't matter which URL I add to Network Service, whenever I add one, and retry the service or restart the WEC, I get the same event ID with a different URL asking me to do the same thing. I've added about 15 URLs before deciding it's just a bug that will continue to recreate itself. Any tips?

On the client side I can see Sysmon64 is running, and the on the wec I can see I have 1 computer active on my subscription. The error was there before I continued to chapter 2, but I decided to try chapter 2 in case that fixed the issue.
I cannot see any logs in the forwarded events section of the event viewer.
Very new to this - please let me know if you need any more information from me!

## To Reproduce
No steps in reproducing, happens every time I restart the subscription. Getting no logs. 

To increase the speed and relevance of the reply we suggest you list down debugging steps you have tried, as well as the following information:
Have tried adding URL's to Network Service - just lists new URL when retrying on Subscriptions.
Have tried adding IPs to netsh http on PowerShell. 
Restarting machines.
Tried quickconfig winrm, says it's already configured.

### Please complete the following information
**Desktop:**
 - OS: Windows 10
 - Browser: Not using browser on DC, using separate laptop to report this. (Using Opera on this laptop)
 - Software version: Haven't been able to install so no software version. 

**Server:** HAVEN'T GOTTEN TO THIS STEP YET, N/A.
- OS: N/A 
WEC and DC on same computer. Windows 10. 

- Relevant container logs: N/A

## Expected behavior
After the first chapter, I expected logs to start showing up from my client laptop, but none showing. Completed second chapter expecting logs to show up, still none.

**Comment by llwaterhouse** [comment]
Hello, sorry to hear you are having issues installing the software. 

We have made a new release since you first reported this issue.  Could you please download the latest release and try again?  

Thank you.

**Comment by aarz-snl** [comment]
Little confused by some of the hardware/software information... are you saying that your WEC/Domain Controller are a windows 10 machine?

Try and post the output of the following: 

winrm get winrm/config

winrm enumerate winrm/config/listener

Are the "checks" at the end of each chapter passing for you? Did you allow 5985 in the firewall? Where is your 'failure event' going to? Is it located in Forwarded Events in event viewer?

**Comment by llwaterhouse** [comment]
If you are still having issues, please notify us by Feb. 17.  After that this issue will be closed.  Thank you.

**Comment by Worldwave** [comment]
Hey there. I am encountering the exact same issue where it re-fails to bind on the WEC every time the service is restarted. I believe something is blocking it from binding to the URLs.

**Comment by aarz-snl** [comment]
This error can be many things from improperly configured GPO's, permissions issues, and incorrect WEC setup using the lme_wec_config.xml 

It's hard to know what is causing both yours particular issues without more information.

**Comment by aarz-snl** [comment]
Additionally which Windows Server are you running? Is it 2012 or 2019? You'll see here it might be different depending on that:

https://learn.microsoft.com/en-US/troubleshoot/windows-server/admin-development/events-not-forwarded-by-windows-server-collector

**Comment by marceloalexei** [comment]
Using 2019. I fixed and have LME completely up and running.
Had to do something with running CMD as admin and adding the WSMAN URL completely. I honestly forgot what the code I used was cause I used so many 'fixes' to finally get to it. All I know is that you have to open CMD as admin and input some command (I found it online at some point) and then it started listening and pushing forward events.

**Comment by aarz-snl** [comment]
And the command was NOT a netsh command? like so:

`netsh http delete urlacl url=http://+:5985/wsman/
netsh http add urlacl url=http://+:5985/wsman/ sddl=D:(A;;GX;;;S-1-5-80-569256582-2953403351-2909559716-1301513147-412116970)(A;;GX;;;S-1-5-80-4059739203-877974739-1245631912-527174227-2996563517)
netsh http delete urlacl url=https://+:5986/wsman/
netsh http add urlacl url=https://+:5986/wsman/ sddl=D:(A;;GX;;;S-1-5-80-569256582-2953403351-2909559716-1301513147-412116970)(A;;GX;;;S-1-5-80-4059739203-877974739-1245631912-527174227-2996563517)`

**Comment by marceloalexei** [comment]
Looked a lot like that actually. Was the difference of using CMD as ADMIN or not. It still 'works' if you use CMD without admin rights but it doesn't actually push anything. 
Additionally I believe I didn't have anything after ...wsman/ , I think I just blanket added that url but it was some weeks ago so maybe I did have those extra numbers/letters after.

**Comment by aarz-snl** [comment]
Ok, thanks... @Worldwave please see that link above for probable resolution. 

The instructions touch on it here:

https://github.com/cisagov/LME/blob/main/docs/markdown/chapter1/chapter1.md#:~:text=%F0%9F%9B%A0%EF%B8%8F%20If%20you%20receive%20the%20error%20%22The,clients%20and%20the%20Windows%20Event%20Collector.

But this may need to be updated to point to multiple types of errors apparently. 

Marceloalexei please let us know of any improvements youd like to see. Ie. "I wish i got these kind of logs" etc

**Comment by marceloalexei** [comment]
Of course. Sorry I can't be of more help.. I may have the exact command on the domain controller back over at the warehouse. Will check in the morning and comment on this if I find it!

**Comment by Worldwave** [comment]
Thanks guys! I was about to comment here earlier that I found that solution last week and it ended up working like a charm. The problem was the SDDL that the automatically-created acl entry included. Changing it to the proper one resolved the issue.
